In [10]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import statistics as stat

In [11]:
df = pd.DataFrame()

In [12]:
df['names'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver']

In [13]:
df['age'] = [14, 12, 11, 10, 8, 6, 8]

In [14]:
print('The mean is {}, the median is {}, and the mode is {}.'.format(np.mean(df['age']), np.median(df['age']), stat.mode(df['age'])))

The mean is 9.857142857142858, the median is 10.0, and the mode is 8.


In [15]:
print('The variance is {}, the standard deviation is {}, and the standard error is {}.'.format(np.var(df.age), np.std(df.age), np.std(df.age) / np.sqrt(len(df.age))))

The variance is 6.408163265306122, the standard deviation is 2.531435020952764, and the standard error is 0.9567925036515135.


## Question 2
For one estimate of central tendency I'd choose the mean since there are no outliers and this is a relatively small data set (I think median is usually more representative on larger data sets). For one estimate of variance I'd choose the standard deviation, because coupling the standard deviation with the mean allows us to calculate summary statistics.

## Question 3

In [23]:
df.loc[5, 'age'] = 7

In [24]:
df

names  age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    7
6  Oliver    8

In [25]:
print('The mean is {}, the median is {}, and the mode is {}.'.format(np.mean(df['age']), np.median(df['age']), stat.mode(df['age'])))

The mean is 10.0, the median is 10.0, and the mode is 8.


In [26]:
print('The variance is {}, the standard deviation is {}, and the standard error is {}.'.format(np.var(df.age), np.std(df.age), np.std(df.age) / np.sqrt(len(df.age))))

The variance is 5.428571428571429, the standard deviation is 2.32992949004287, and the standard error is 0.8806305718527109.


The mean, variance, std deviation, and std error changed but the median and mode did not.

## Question 4

In [27]:
df.loc[6, 'names'] = 'Jessica'

In [28]:
df.loc[6, 'age'] = 1

In [29]:
df

names  age
0     Greg   14
1   Marcia   12
2    Peter   11
3      Jan   10
4    Bobby    8
5    Cindy    7
6  Jessica    1

In [31]:
print('The mean is {}, the median is {}, and the mode is {}.'.format(np.mean(df['age']), np.median(df['age']), 'not meaningful'))

The mean is 9.0, the median is 10.0, and the mode is not meaningful.


In [32]:
print('The variance is {}, the standard deviation is {}, and the standard error is {}.'.format(np.var(df.age), np.std(df.age), np.std(df.age) / np.sqrt(len(df.age))))

The variance is 15.428571428571429, the standard deviation is 3.927922024247863, and the standard error is 1.4846149779161806.


I would still choose mean and standard deviation because there are still no outliers and these two allow us to calculate summary statistics.

## Question 5

In [33]:
fan_list = [20, 23, 17, 5]

In [35]:
df2 = pd.DataFrame()

In [36]:
df2['percentages'] = fan_list

In [37]:
df2.percentages.describe()

count     4.000000
mean     16.250000
std       7.889867
min       5.000000
25%      14.000000
50%      18.500000
75%      20.750000
max      23.000000
Name: percentages, dtype: float64

My best guess is that 16.25% of the adult population were Brady Bunch fans, with 68% confidence that the answer lies between 8.36%-25.14% and 95% confidence that the answer lies between 0.47%-33.03%.